pizza_cutter/des_pizza_cutter/_coadd_slices.py

```diff
@@ -555,3 +567,132 @@ def _coadd_slice_inputs(
         interp_se_frac,
         resampled_datas,
     )
+
+
+@functools.lru_cache(maxsize=1)
+def _get_gaia_stars(*, fname):
+    """
+    load the gaia stars
+
+    Parameters
+    -----------
+    fname: str
+        path to the gaia star catalog
+    """
+    full_path = os.path.expandvars(fname)
+    logger.info('reading: %s' % full_path)
+    data = fitsio.read(full_path, lower=True)
+    data = eu.numpy_util.add_fields(data, [('radius_arcsec', 'f4')])
+
+    ply = np.poly1d([0.00443223, -0.22569131, 2.99642999])
+
+    # these do not have a radius factor in them
+    log10_radius_arcsec = ply(data['phot_g_mean_mag'])
+    data['radius_arcsec'] = 10.0**log10_radius_arcsec
+    return data
+
+
+@njit
+def _intersects(row, col, radius_pixels, nrows, ncols):
+    """
+    low level routine to check if the mask intersects the image.
+    For simplicty just check the bounding rectangle
+
+    Parameters
+    ----------
+    row, col: float
+        The row and column of the mask center
+    radius_pixels: float
+        The radius for the star mask
+    nrows, ncols: int
+        Shape of the image
+
+    Returns
+    -------
+    True if it intersects, otherwise False
+    """
+
+    low_row = -radius_pixels
+    high_row = nrows + radius_pixels - 1
+    low_col = -radius_pixels
+    high_col = ncols + radius_pixels - 1
+
+    if (
+        row > low_row and row < high_row and
+        col > low_col and col < high_col
+    ):
+        return True
+    else:
+        return False
+
+
+@njit
+def _do_mask_gaia_stars(rows, cols, radius_pixels, bmask, flag):
+    """
+    low level code to mask stars
+
+    Parameters
+    ----------
+    rows, cols: arrays
+        Arrays of rows/cols of star locations; may be off the image
+    radius_pixels: float
+        The radius for each star mask
+    bmask: array
+        The bmask to modify
+    flag: int
+        The flag value to "or" into the bmask
+    """
+    nrows, ncols = bmask.shape
+
+    for istar in range(rows.size):
+        row = rows[istar]
+        col = cols[istar]
+        rad = radius_pixels[istar]
+        rad2 = rad * rad
+
+        if not _intersects(row, col, rad, nrows, ncols):
+            continue
+
+        for irow in range(nrows):
+            rowdiff2 = (row - irow)**2
+            for icol in range(ncols):
+
+                r2 = rowdiff2 + (col - icol)**2
+                if r2 < rad2:
+                    bmask[irow, icol] |= flag
+
+
+def _mask_gaia_stars(gaia_stars, wcs, bmask, radius_factor=1, maxmag=18):
```

Comment:
the radius factor and maxmag should be in the config file too.

Reply:
I think the way the code is structured,these would have to be keywords to the `_coadd_and_write_images` and `_coadd_slice_inputs` functions.   Is that right?

Reply:
Or you can pass down an opaque dictionary of options that matches a yaml dict in the config. I'd be fine with that.